### PR TITLE
[RFR] Admin page to manage removed resources

### PR DESCRIPTION
**Pull request by djhi**
[Trello card](https://trello.com/c/J8411QYF)

### CODE REVIEW

File: src/api/controller/api/publishedDataset.js [hunk 1]
```diff
@@ -10,7 +10,26 @@ export const getPage = async (ctx) => {
...
+    };
+};
+
+export const getRemovedPage = async (ctx) => {
+    const { page = 0, perPage = 10 } = ctx.request.query;
+    const intPage = parseInt(page, 10);
+    const intPerPage = parseInt(perPage, 10);
+
+    const [data, total] = await Promise.all([
+        ctx.publishedDataset.findPage(intPage, intPerPage, true), // true for removed
```
Comment (ThieryMichel) [inline src/api/controller/api/publishedDataset.js]: create and use findRemovedPage

File: src/api/controller/api/publishedDataset.js [hunk 2]
```diff
@@ -10,7 +10,26 @@ export const getPage = async (ctx) => {
...
+        ctx.publishedDataset.countWithRemoved(true), // true for removed
```
Comment (ThieryMichel) [inline src/api/controller/api/publishedDataset.js]: countRemoved

File: src/api/models/publishedDataset.js [hunk 3]
```diff
@@ -10,14 +10,17 @@ export default (db) => {
     collection.findLimitFromSkip = (limit, skip, filter) =>
         collection.find(filter).skip(skip).limit(limit).toArray();
 
-    collection.findPage = (page = 0, perPage = 10) =>
-        collection.findLimitFromSkip(perPage, page * perPage, { removedAt: { $exists: false } });
+    collection.findPage = (page = 0, perPage = 10, removed) =>
```
Comment (ThieryMichel) [inline src/api/models/publishedDataset.js]: Add a findRemovedPage calling findPage with removed = true.

File: src/api/models/publishedDataset.js [hunk 4]
```diff
@@ -10,14 +10,17 @@ export default (db) => {
...
+        collection.findLimitFromSkip(perPage, page * perPage, { removedAt: { $exists: removed } });
+
+    collection.countWithRemoved = removed =>
```
Comment (ThieryMichel) [inline src/api/models/publishedDataset.js]: Add countRemoved and countNotRemoved

File: src/app/js/admin/removedResources/index.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+export const defaultState = {
+    currentPage: 0,
+    error: false,
+    loading: false,
+    items: [],
+    total: 0,
+};
+
+export default handleActions({
+    [combineActions(LOAD_REMOVED_RESOURCE_PAGE, RESTORE_RESOURCE)]: state => ({
```
Comment (ThieryMichel) [inline src/app/js/admin/removedResources/index.js:28]: :+1:

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: src/api/controller/api/publishedDataset.js (hunk 2 above)
Comment (djhi) [reply src/api/controller/api/publishedDataset.js]: Well, I'll have to introduce `countWithoutRemoved` then

**Review by ThieryMichel — APPROVED** [review]

File: src/app/js/admin/removedResources/index.js (hunk 5 above)
Comment (djhi) [reply src/app/js/admin/removedResources/index.js:28]: yea, redux-actions rocks :)

**Pull request merged**